nomad/structs/keyring.go

```diff
@@ -243,3 +253,38 @@ type KeyringDeleteRootKeyRequest struct {
 type KeyringDeleteRootKeyResponse struct {
 	WriteMeta
 }
+
+// KeyringListPublicResponse lists public key components of signing keys. Used
+// to build a JWKS endpoint.
+type KeyringListPublicResponse struct {
+	PublicKeys []*KeyringPublicKey
+
+	// RotationThreshold exposes root_key_rotation_threshold so that HTTP
+	// endpoints may set a reasonable cache control header informing consumers
+	// when to expect a new key.
+	RotationThreshold time.Duration
+
+	QueryMeta
+}
+
+// KeyringPublicKey is the public key component of a signing key. Used to build
+// a JWKS endpoint.
+type KeyringPublicKey struct {
+	KeyID     string
+	PublicKey []byte
+
+	// Algorithm should be the JWA "alg" parameter. So "EdDSA" for Ed25519 public
```

Comment:
```suggestion
	// Algorithm should be the JWT "alg" parameter. So "EdDSA" for Ed25519 public
```

I think?

Reply:
I think technically this is a correct statement:

> The JWA algorithm EdDSA is used in the JWS header of the JWT.

https://datatracker.ietf.org/doc/html/rfc8037#appendix-A.4

That being said in Nomad's context we're always talking about JWTs, so I think just using "JWT" everywhere makes more sense than trying to tease apart exactly which layer of the JOSE stack each little parameter applies to... ...at least until we add [x509](https://www.rfc-editor.org/rfc/rfc8446.html#section-4.2.3) support. :grin:

Reply:
> The JWA algorithm EdDSA is used in the JWS header of the JWT.

ha! TIL! :grinning: